component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/EmailOtpServiceImpl.java

```diff
@@ -147,6 +166,25 @@ public GenerationResponseDTO generateEmailOTP(String userId, String emailOtpExpi
                         String.format("Error while retrieving user for the Id : %s.", userId), e);
             }
 
+            boolean showFailureReason = EmailOtpServiceDataHolder.getConfigs().isShowFailureReason();
```

Comment:
Shall we move this config to class level and use static final variable

Reply:
fixed with e31be4c77fdedf379c272da521b6e2ab837b2a11